samples/sample_common/include/sample_utils.h

```diff
@@ -540,6 +540,37 @@ inline void ExtBufHolder<mfxFrameSurface1>::RefreshBuffers()
     this->Data.ExtParam    = this->Data.NumExtParam ? m_ext_buf.data() : nullptr;
 }
 
+template <>
+inline ExtBufHolder<mfxFrameSurface1>& ExtBufHolder<mfxFrameSurface1>::operator=(const mfxFrameSurface1& ref)
+{
+    // copy content of main structure type T
+    mfxFrameSurface1* dst_base = this;
+    const mfxFrameSurface1* src_base = &ref;
+    *dst_base = *src_base;
```

Comment:
I believe you don't need all those pointers, just do
```
*this = ref;
```

Reply:
I agree. And you don't need mfxFrameSurface1 specialization. Make it generic.

Reply:
@mgonchar  thanks.
@dmitryermilov in this specific case  `mfxFrameSurface1` has no `NumExtParam` and `*ExtParam`, because they are contains in `Data` field. In this reason I add this specialization.

Reply:
@AntonGrishin ok, it's easy to fix:
```
static_cast<mfxFrameSurface1&>(*this) = ref;
```

Reply:
[diff.txt](https://github.com/Intel-Media-SDK/MediaSDK/files/3635948/diff.txt)
Please try this

Reply:
@dmitryermilov applied with some changes.
@mgonchar Function removed.